File: plugin/zeppelin/src/main/scala/jetbrains/zeppelin/service/FileNotebookHolder.scala
package jetbrains.zeppelin.service

import com.intellij.openapi.extensions.ExtensionPointName
import com.intellij.psi.PsiFile
import jetbrains.zeppelin.constants.ZeppelinConstants

/**
  * Extension point which define a notebook id fore current opened file
  */
trait FileNotebookHolder {
  def contains(psiFile: PsiFile): Boolean

  def notebookId(psiFile: PsiFile): String
}

object FileNotebookHolder {
  val Id: String = ZeppelinConstants.PLUGIN_ID + ".fileNotebookHolder"
  val EP_NAME: ExtensionPointName[FileNotebookHolder] =
    ExtensionPointName.create[FileNotebookHolder](Id)

  def getAll: Array[FileNotebookHolder] = EP_NAME.getExtensions
}

File: zeppelin/implementation/src/org/intellij/plugin/zeppelin/extensionpoints/FileNotebookHolder.kt
package org.intellij.plugin.zeppelin.extensionpoints

import com.intellij.openapi.extensions.ExtensionPointName
import com.intellij.psi.PsiFile
import org.intellij.plugin.zeppelin.constants.ZeppelinConstants

/**
 * Extension point which define a notebook id fore current opened file
 */
interface FileNotebookHolder {
    fun contains(psiFile: PsiFile): Boolean
    fun getNotebookId(psiFile: PsiFile): String

    companion object {
        private val Id: String = ZeppelinConstants.PLUGIN_ID + ".fileNotebookHolder"
        private val EP_NAME: ExtensionPointName<FileNotebookHolder> = ExtensionPointName.create<FileNotebookHolder>(
                Id)

        fun getAll(): List<FileNotebookHolder> = EP_NAME.extensions.toList()
    }
}
